### PR TITLE
Improve lint script

### PR DESCRIPTION
**Pull request by fisker**
Skip `outputFixes` and `report` when there is no error.

Add tips for how to fix.

### CODE REVIEW

File: test/lint/lint.js [hunk 1]
```diff
@@ -24,12 +21,30 @@ cli.addPlugin('eslint-plugin-unicorn', unicorn);
...
 }
 
-const formatter = cli.getFormatter();
+if (errorCount || warningCount) {
+	const formatter = cli.getFormatter();
+	console.log(formatter(report.results));
 
-console.log(formatter(report.results));
+	console.log();
+	console.log('Some test fails, you need fix them, and run `npm run lint <file>` again.');
```
Comment (yakov116) [inline test/lint/lint.js]:
```suggestion
	console.log('Some test have failed, to fix them run `npm run lint <file>` again.');
```

File: test/lint/lint.js [hunk 2]
```diff
@@ -24,12 +21,30 @@ cli.addPlugin('eslint-plugin-unicorn', unicorn);
...
+
+	if (hasFixable) {
+		console.log();
+		console.log('You may also want run `npm run lint <file> --fix` to fix fixable problems.');
+	}
+
+	console.log();
+	console.log('* If you\'re making a new rule, you can fix this later. *');
+} else {
+	console.log('All test passed.');
```
Comment (yakov116) [inline test/lint/lint.js]:
```suggestion
	console.log('All test have passed.');
```

File: test/lint/lint.js (hunk 1 above)
Comment (fisker) [reply test/lint/lint.js]: `npm run lint <file>` can't fix them, need fix and run to check

Comment (yakov116) [reply test/lint/lint.js]: Would `to what needs to be fixed`... Be better?

Comment (fisker) [reply test/lint/lint.js]: noop, what needs to fix is already listed.

how about 

> Some tests have failed, you need fix them and run `npm run lint <file>` to check again.

Comment (yakov116) [reply test/lint/lint.js]: @fisker aha! Yes that sounds good. (I misunderstood what the message was supposed to say 😂)

File: test/lint/lint.js [hunk 3]
```diff
@@ -22,14 +19,34 @@ const cli = new CLIEngine({
 
 cli.addPlugin('eslint-plugin-unicorn', unicorn);
 
+// Find a way to make sure rules are loaded from codebase
```
Comment (fisker) [inline test/lint/lint.js]: Hard to to this, https://github.com/eslint/eslint/issues/12425 , will find a way

File: test/lint/lint.js [hunk 4]
```diff
@@ -22,14 +19,34 @@ const cli = new CLIEngine({
...
 }
 
-const formatter = cli.getFormatter();
+if (errorCount || warningCount) {
+	const formatter = cli.getFormatter();
+	console.log(formatter(report.results));
 
-console.log(formatter(report.results));
+	console.log();
+	console.log('Some tests have failed, you need fix them and run `npm run lint <file>` to check again.');
```
Comment (yakov116) [inline test/lint/lint.js]: what about `Not all tests have passed`?
Just a thought since it can be only one test that failed

Comment (fisker) [reply test/lint/lint.js]: haha, okay. I can also know how many (errorCount + warningCount)

Comment (fisker) [reply test/lint/lint.js]: @yakov116

https://travis-ci.org/sindresorhus/eslint-plugin-unicorn/jobs/604272880#L225

good?

**Comment by fisker** [comment]
Ready.

File: test/lint/lint.js [hunk 5]
```diff
@@ -22,14 +21,39 @@ const cli = new CLIEngine({
...
 	CLIEngine.outputFixes(report);
 }
 
-const formatter = cli.getFormatter();
+if (errorCount || warningCount) {
+	const formatter = cli.getFormatter();
+	console.log(formatter(report.results));
+
+	console.log();
+	console.log(`You need fix failed test${errorCount + warningCount > 1 ? 's' : ''} above and run \`npm run lint <file>\` to check again.`);
```
Comment (yakov116) [inline test/lint/lint.js]:
```suggestion
	console.log(`You need to fix the failed test${errorCount + warningCount > 1 ? 's' : ''} above and run \`npm run lint <file>\` to check again.`);
```

File: test/lint/lint.js (hunk 4 above)
Comment (yakov116) [reply test/lint/lint.js]: Perfect!

File: docs/new-rule.md [hunk 6]
```diff
@@ -24,6 +24,6 @@ Use the [`astexplorer` site](https://astexplorer.net) with the `espree` parser a
 	*(The description should be the same as the heading of the documentation file).*
 - Run `$ npm test` to ensure the tests pass.
 - Run `$ npm run integration` to run the rules against real projects to ensure your rule does not fail on real-world code.
-- Run `$ npm run lint` to run the rules against this repository to ensure code in the repository are following your rule.
 - Open a pull request with a title in exactly the format `` Add `rule-name` rule ``, for example, `` Add `no-unused-properties` rule ``.
 - The pull request description should include the issue it fixes, for example, `Fixes #123`.
+- Run `$ npm run lint` to run the rules against codebase to ensure code in the repository are following your rule, _you can ignore this step until your PR is reviewed_.
```
Comment (fisker) [inline docs/new-rule.md]: @yakov116 any suggestion?

Comment (yakov116) [reply docs/new-rule.md]: Not sure maybe 

to run the rules against the codebase. Ensuring that the rest of the codebase follows your rule.

Comment (fisker) [reply docs/new-rule.md]: not rest, new rule self may also not pass, should be all files

**Pull request merged**